lib/ui/painting/image_decoder_unittests.cc

```diff
@@ -197,9 +207,10 @@ TEST_F(ImageDecoderFixtureTest, ValidImageResultsInSuccess) {
     runners.GetUITaskRunner()->PostTask(decode_image);
   };
 
+  EXPECT_FALSE(io_manager->did_access_is_gpu_disabled_sync_switch_);
   runners.GetIOTaskRunner()->PostTask(setup_io_manager_and_decode);
-
   latch.Wait();
+  EXPECT_TRUE(io_manager->did_access_is_gpu_disabled_sync_switch_);
```

Comment:
Couldn't you mock/fake out your own implementation of `SyncSwitch` here so that you could also assert that it had `Execute` called on it (and maybe which branch)?

Reply:
Not without making `Execute` virtual.

Reply:
Ahh ok.  This is probably fine then.